master/buildbot/util/sautils.py

```diff
@@ -58,3 +59,16 @@ def Table(*args, **kwargs):
     # a non-utf8 character set (mysql's default)
     kwargs['mysql_character_set'] = 'utf8'
     return sa.Table(*args, **kwargs)
+
+
+@contextmanager
+def withoutSqliteForeignKeys(engine):
+    if engine.dialect.name == 'sqlite':
+        engine.fk_disabled = True
```

Comment:
Maybe assert here that `engine.fk_disabled` was `False`?
I don't say that is worth it to make `withoutSqliteForeignKeys` reentrant, but if not at least assert that it's not called recursively should be added.